src/posit/connect/permissions.py

```diff
@@ -137,3 +148,86 @@ def get(self, uid: str) -> Permission:
+            The content item permissions to remove. If a `str` is received, it is compared against
+            the Permissions' `principal_gruid`. If a `Group`, `User`, or `Permission` is received,
+            the `guid` is used and compared against the Permissions' `principal_guid`. If a
+            `Permission` is received, the `principal_guid` is used. Note, only the associated permissions will be destroyed; Any users or groups provided will remain.
+
+        Returns
+        -------
+        list[Permission]
+            The deleted permissions.
+
+        Examples
+        --------
+        ```python
+        from posit import connect
+
+        #### User-defined inputs ####
+        # 1. specify the guid for the content item
+        content_guid = "CONTENT_GUID_HERE"
+        # 2. specify either the principal_guid or group name prefix
+        principal_guid = "USER_OR_GROUP_GUID_HERE"
+        group_name_prefix = "GROUP_NAME_PREFIX_HERE"
+        ############################
+
+        client = connect.Client()
+
+        # Remove a single permission by principal_guid
+        client.content.get(content_guid).permissions.destroy(principal_guid)
+
+        # Remove by user (if principal_guid is a user)
+        user = client.users.get(principal_guid)
+        client.content.get(content_guid).permissions.destroy(user)
+
+        # Remove by group (if principal_guid is a group)
+        group = client.groups.get(principal_guid)
+        client.content.get(content_guid).permissions.destroy(group)
+
+        # Remove all groups with a matching prefix name
+        groups = client.groups.find(prefix=group_name_prefix)
+        client.content.get(content_guid).permissions.destroy(*groups)
+
+        # Confirm new permissions
+        client.content.get(content_guid).permissions.find()
+        ```
+        """
+        from .groups import Group
+        from .users import User
+
+        if len(permissions) == 0:
+            raise ValueError("Expected at least one `permission` to remove")
+
+        principal_guids: set[str] = set()
+
+        for arg in permissions:
+            if isinstance(arg, str):
+                principal_guid = arg
+            elif isinstance(arg, Group):
+                principal_guid: str = arg["guid"]
+            elif isinstance(arg, User):
+                principal_guid: str = arg["guid"]
+            elif isinstance(arg, Permission):
+                principal_guid: str = arg["principal_guid"]
+            else:
+                raise TypeError(
+                    f"destroy() expected argument type 'str', 'Group', 'Permission' or 'User', but got '{type(arg).__name__}'",
+                )
+            principal_guids.add(principal_guid)
+
+        destroyed_permissions: list[Permission] = []
+        for permission in self.find():
+            if permission["principal_guid"] in principal_guids:
+                permission.destroy()
+                destroyed_permissions.append(permission)
```

Comment:
Am I following correctly that this will silently skip any Users/Groups/Permissions that are in the args, but aren't already on the piece of content? That's probably the right behavior, but should we add that to the documentation? Are there edge cases we want to catch and warn about (i.e. there are no matches at all)?

Reply:
Correct, mismatches will be silent. 

I'll add that to the docs.

I'd like to not submit warnings if possible. My interpretation is that at the end of the function, all associated provided permissions will not exist. The permissions that were removed are returned so users can make their own assertions / warnings.